pkg/distributor/forwarding/forwarding.go

```diff
@@ -84,6 +86,11 @@ func NewForwarder(reg prometheus.Registerer, cfg Config) Forwarder {
 			Name:      "distributor_forward_requests_total",
 			Help:      "The total number of requests the Distributor made to forward samples.",
 		}, []string{"user"}),
+		errorsTotal: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Namespace: "cortex",
+			Name:      "distributor_forward_errors_total",
+			Help:      "The total number of errors which the Distributor received from forwarding targets.",
+		}, []string{"user", "status_code"}),
```

Comment:
Do you need it on a per-user basis? Asking because this is a good candidate for an high cardinality metric when running at scale.

Reply:
Good question, I think an alternative would be to do key it by the forwarding target instead of the user, that would give us all the information we need to detect if a forwarding target doesn't function as expected. 

In the way we use this functionality at the moment this won't make a difference for the cardinality because there is a 1:1 mapping between users and forwarding targets. However, in the future I think the cardinality might be lower if we key it by the forwarding target.

Reply:
Updated it here: https://github.com/grafana/mimir/pull/2077/commits/21cfb8d527bed849679d43771e55794ea49d1f3e

Reply:
Actually, I changed my mind, I can't think of a reason why we need any of these on a per-user basis... I'll remove the user label and the target label from all of them, if we find a reason why we do need them we can still add them later.

Reply:
Updated again, removed all labels except the `status_code` on `distributor_forward_errors_total`. I'm going to merge this now to move forward, if we still need to make other changes I'm happy to create more PRs